cloudinit/reporting/__init__.py

```diff
@@ -32,19 +34,19 @@ def update_configuration(config):
             )
             continue
         handler_config = handler_config.copy()
-        cls = available_handlers.registered_items[handler_config.pop("type")]
+        cls: Type[HandlerType] = available_handlers.registered_items[
+            handler_config.pop("type")
+        ]
         instantiated_handler_registry.unregister_item(handler_name)
-        instance = cls(**handler_config)
+        instance = cls(**handler_config)  # pyright: ignore
```

Comment:
requires pyright >= 1.1.229

Reply:
Since it's a comment, it shouldn't require pyright at all, right? We already have a handful of pyright comments other places.

Reply:
No need for change here, this was more a note to self than anything I guess. I just was wondering why pyright wasn't respecting the comment locally and realized an upgrade was required to pick up the change.